docs/quickstarts/csharp-quickstart.md

```diff
@@ -0,0 +1,254 @@
+
+```sh
+$ # Clone the repository to get the example code:
+$ git clone -b $(curl -L http://grpc.io/release) https://github.com/grpc/grpc 
+```
+
+All of the files in this quickstart are in the directory `examples/csharp/helloworld`.
+
+## Build the example
+
+* Open solution `Greeter.sln` with Visual Studio, Monodevelop (on Linux) or Xamarin Studio (on Mac OS X)
+* Build the solution (this will automatically download NuGet dependencies)
+  
+## Run a gRPC application
+
+From the `examples/csharp/helloworld` directory:
+
+1. Run the server
+
+```
+> cd GreeterServer/bin/Debug
+> GreeterServer.exe
+```
+
+2. In another terminal, run the client
+
+```
+> cd GreeterClient/bin/Debug
+> GreeterClient.exe
+```
+
+Note to run the above executables with "mono" if building on Xamarin Studio for Mac OS X.
+
+Congratulations! You've just run a client-server application with gRPC.
+
+## Update a gRPC service
+
+Now let's look at how to update the application with an extra method on the
+server for the client to call. Our gRPC service is defined using protocol
+buffers; you can find out lots more about how to define a service in a `.proto`
+file in [What is gRPC?]() and [gRPC Basics: C#][]. For now all you need
+to know is that both the server and the client "stub" have a `SayHello` RPC
+method that takes a `HelloRequest` parameter from the client and returns a
+`HelloResponse` from the server, and that this method is defined like this:
+
+
+```
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+Let's update this so that the `Greeter` service has two methods. Edit `examples/proto/helloworld.proto` and update it with a new `SayHelloAgain` method, with the same request and response types:
+
+```
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+  // Sends another greeting
+  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+(Don't forget to save the file!)
+
+## Generate gRPC code
+
+Next we need to update the gRPC code used by our application to use the new service definition. 
+The Grpc.Tools NuGet package contains the protoc and protobuf C# plugin binaries you will need to generate the code. This example project already depends on the nuget package `Grpc.Tools.0.15.0`, so it should be included in the `examples/csharp/helloworld/packages` when the `Greeter.sln` solution is built from your IDE. 
```

Comment:
Since the commit is to the GA branch which will only be 'visible' to users after we decide to make it the master branch, we will not have a situation where the user will see the examples while the 0.15 version is the latest.

Therefore, to simplify things, you can assume that this branch contains documentation that users will see once the 1.0 packages are out.